libshaderc_util/src/compiler.cc

```diff
@@ -140,7 +151,7 @@ bool Compiler::Compile(
                                    /* suppress_warnings = */ true,
                                    glslang_errors.c_str(), total_warnings,
                                    total_errors);
-    if (!success) return false;
+    if (!success) return std::make_tuple(false, std::vector<uint32_t>(), 0u);
```

Comment:
This std::make_tuple expression is repeated a lot.
You could make a const auto error_result = std::tuple(false, std::vector<uint32_t>(), 0u) and then use that all over the place.